test/client.test.js

```diff
@@ -143,6 +143,29 @@ describe('Client', () => {
     return expect(resultWithoutOptions).toEqual(resultWithOptions)
   })
 
+  test('cleans up http2 sessions when .close is called', async () => {
```

Comment:
Any way we can test the behavior for both HTTP1 and HTTP2? We have changed the behavior for HTTP1 so we should test the new semantics (no requests after close).